### ISSUE TITLE
Rule proposal: `no-await-only-function`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1341]
If there is only one `return await` in function, should prefer return Promise. Better rule name?

## Fail

```js
const foo = async () => await bar();
```

```js
const foo = async () => {
	return await bar();
}
```

```js
async function foo() {
	return await bar();
}
```

## Pass

```js
const foo = () => bar();
```

```js
const foo = async () => {
	await bar();
}
```

```js
function foo() {
	return bar();
}
```

**Comment by fisker** [comment]
Here is another similar case, https://github.com/sindresorhus/eslint-plugin-unicorn/blob/089785af133f0b24dafa8c489925b0954ed5cb6c/rules/new-for-builtins.js#L57-L59

I wrote this, but I don't know why...

## Fail

```js
function foo * () {
	yield * bar()
}
```

## Pass

```js
function foo () {
	return bar()
}
```

```js
function * foo () {
	yield bar();
      // ^ no `*` here
}
```

It's like a "yield-star-only-function".

**Comment by fisker** [comment]
This is already covered by [`no-return-await` rule](https://eslint.org/docs/rules/no-return-await)
